### DISCUSSION TITLE
Tests using beanie models

### DISCUSSION THREAD

**Discussion by yuval9313** [BeanieODM/beanie#545] [Question]
Hello,

I wish to test using beanie as it represents the models in my system.

With beanie I've expected the same behavior like the JS lib mongoose
1. The models declared in beanie are the DTOs I use across the app
2. I use the repository layer to hide DB operations

And then I can mock the repository layer and prevent the need to use actual DB

However, when attempting this in beanie I stumble upon `CollectionWasNotInitialized`
Which seems weird since I didn't expect the Model to reach the DB unless I actually call `beanie.init`

Could you suggest a possible solution? I don't want to fire up in-memory/docker instance of mongodb...

**Comment by roman-right** [comment]
Hi. Currently, there is no solution for this. The thing is on the init step it prepares a lot of things on the class level - without them most of the methods, or will not work, or will work incorrectly. If you want to use mocks, you probably can take a look at this lib: https://github.com/roman-right/beanita. Be aware - it doesn't cover some mongodb operations.
I'll think about adding a db-less mode for the models. But I can not estimate when it will be done.

**Reply by yuval9313** [reply]
I'm happy beanita is a thing, but I'm worried it would replace the no-db-model approach presented in similar ODM

I really like this lib, and it is really easy and enabling so I would like to help with the approach mentioned (or perhaps any other approach that simplify the DTO usage)

But I'd love guidance in that subject so we can reach a comprehensive solution.

P.S - IMO ideally, late startup/methods that restricted to init could raise an exception, while simply defining Model shouldn't, I'll check myself if I can make this happen

**Reply by piaria** [reply]
Hi, 

I write my tests with pytest using `mongomock`  and calling `init_beanie` this way:
```python
async def initialize_bd():
    client = AsyncMongoMockClient()
    await init_beanie(
        document_models=app_models + emodelslib_models,
        database=client.get_database(name="test_db"),
    )
```

best,
Pablo.-
